application/wypl-core/src/main/java/com/wypl/wyplcore/review/service/ReviewServiceImpl.java

```diff
@@ -0,0 +1,211 @@
+import com.wypl.wyplcore.review.exception.ReviewException;
+import com.wypl.wyplcore.review.utils.ReviewUtils;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class ReviewServiceImpl implements ReviewModifyService, ReviewReadService {
+	private final ReviewRepository reviewRepository;
+
+	// private final ScheduleRepository scheduleRepository;
+
+	private final ReviewContentsRepository reviewContentsRepository;
+
+	@Override
+	@Transactional
+	public ReviewIdResponse createReview(int memberId, ReviewCreateRequest reviewCreateRequest) {
+		validateReviewContents(reviewCreateRequest.contents(), reviewCreateRequest.title());
+
+		//schedule, member 유효성 판단
+		// MemberSchedule memberSchedule = memberScheduleService.getMemberScheduleByMemberAndSchedule(memberId,
+		// 	ScheduleServiceUtils.findById(scheduleRepository, reviewCreateRequest.scheduleId()));
+		// memberSchedule.writeReview();
+
+		Member findMember = null;    // FIXME: MemberUtil
+		Schedule findSchedule = null;    // FIXME: ScheduleUtil
+
+		Review review = Review.of(reviewCreateRequest.title(), findMember, findSchedule);
+		Review savedReview = reviewRepository.save(review);
+
+		reviewContentsRepository.save(ReviewContents.of(savedReview.getReviewId(), reviewCreateRequest.contents()));
+		return ReviewIdResponse.from(savedReview.getReviewId());
+	}
+
+	@Override
+	@Transactional
+	public ReviewIdResponse updateReview(int memberId, int reviewId, ReviewUpdateRequest reviewUpdateRequest) {
+		validateReviewContents(reviewUpdateRequest.contents(), reviewUpdateRequest.title());
+
+		Review review = ReviewUtils.findReviewByReviewIdAndMemberId(reviewRepository, reviewId, memberId);
+		review.updateTitle(reviewUpdateRequest.title());
+
+		ReviewContents reviewContents = reviewContentsRepository.findByReviewIdAndDeletedAtNull(review.getReviewId());
+		reviewContents.updateContents(reviewUpdateRequest.contents());
+
+		reviewContentsRepository.save(reviewContents);
+
+		return ReviewIdResponse.from(review.getReviewId());
+	}
+
+	@Override
+	@Transactional
+	public ReviewIdResponse deleteReview(int memberId, int reviewId) {
+		Review review = ReviewUtils.findReviewByReviewIdAndMemberId(reviewRepository, reviewId, memberId);
+
+		ReviewContents reviewContents = reviewContentsRepository.findByReviewIdAndDeletedAtNull(review.getReviewId());
+		reviewContents.delete();
+
+		reviewContentsRepository.save(reviewContents);
+
+		review.delete();
+
+		return ReviewIdResponse.from(review.getReviewId());
+	}
+
+	@Override
+	public ReviewDetailResponse getDetailReview(long memberId, long reviewId) {
+		Review review = ReviewUtils.findReviewByReviewIdAndMemberId(reviewRepository, reviewId, memberId);
+		Schedule schedule = review.getSchedule();
+
+		// Todo : 저장할 때 null 검사 하는데, 가져올 때도 검사해야 할까???
+		ReviewContents reviewContents = reviewContentsRepository.findByReviewIdAndDeletedAtNull(review.getReviewId());
```

Comment:
음.. 제 생각으로는 가져올 때 Null 검사를 한다면 가져올 때 검사를 할 필요는 없다고 생각합니다!
```suggestion
// Todo : 저장할 때 null 검사 하는데, 가져올 때도 검사해야 할까???
ReviewContents reviewContents = reviewContentsRepository.findByReviewIdAndDeletedAtNull(review.getReviewId());
```